packages/doenetml-worker-rust/lib-doenetml-core/src/core/state_var_interfaces/boolean_state_var_interfaces.rs

```diff
@@ -62,30 +65,34 @@ impl TryFrom<&StateVarReadOnlyViewEnum> for BooleanOrString {
     }
 }
 
-// impl Default for BooleanOrStrings {
-//     fn default() -> Self {
-//         BooleanOrStrings::Boolean(StateVarReadOnlyView::default())
-//     }
-// }
+impl GeneralBooleanStateVarInterface {
+    pub fn new(base_dependency_instruction: DependencyInstruction) -> Self {
+        GeneralBooleanStateVarInterface {
+            base_dependency_instruction,
+            ..Default::default()
+        }
+    }
+
+    /// Set parameter `is_primary_state_variable` to true
+    pub fn is_primary_state_variable(mut self) -> Self {
```

Comment:
I think non-mutable would be better.

Reply:
You need to recreate the whole object.

Reply:
I could clone it and then set the new parameter, but since that wouldn't be a good idea if one had large objects.

Is it better to recreate by specifying all the fields again like below? Seems simpler to mutate the original object, given that you've taken ownership of it anyway.

```
        GeneralStringStateVarInterface {
            is_primary_state_variable: true,
            base_dependency_instruction: self.base_dependency_instruction,
            dependency_instructions: self.dependency_instructions,
            dependency_values: self.dependency_values,
            from_single_essential: self.from_single_essential,
        }
```

Reply:
Yeah. Recreating it is the proper way. The compiler should optimize an inefficiency away.

Reply:
OK. It became a moot point, though, as I eliminated the primary state variable.